File: app/models/athlete.py
import uuid
from typing import Optional
from datetime import datetime

from sqlalchemy import String, DateTime, Float, Integer, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Athlete(Base):
    __tablename__ = "athletes"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)

    organization_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("organizations.id", ondelete="CASCADE")
    )

    first_name: Mapped[str] = mapped_column(String(120))
    last_name: Mapped[str] = mapped_column(String(120))

    weight_kg: Mapped[Optional[float]] = mapped_column(Float)
    height_cm: Mapped[Optional[float]] = mapped_column(Float)

    ftp_watts: Mapped[Optional[int]] = mapped_column(Integer)
    threshold_hr: Mapped[Optional[int]] = mapped_column(Integer)

    vo2max: Mapped[Optional[float]] = mapped_column(Float)

    garmin_access_token: Mapped[Optional[str]] = mapped_column(String(500))

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

File: noa-frontend/src/types/athlete.ts
export type Athlete = {
  id: string;
  organization_id: string;
  first_name: string;
  last_name: string;
  weight_kg: number | null;
  height_cm: number | null;
  ftp_watts: number | null;
  threshold_hr: number | null;
  vo2max: number | null;
  created_at: string;
};

export type AthleteCreatePayload = {
  organization_id: string;
  first_name: string;
  last_name: string;
  weight_kg: number | null;
  height_cm: number | null;
  ftp_watts: number | null;
  threshold_hr: number | null;
  vo2max: number | null;
};
